### PR TITLE
Added support for CZML rectangles

### PR DESCRIPTION
**Pull request by idanmiara**
core.py - added rectangle attribute to the CZML Packet class
properties.py - added CartographicRectangle, RectangleCoordinates CZML object classes

### CODE REVIEW

**Comment by idanmiara** [comment]
Hi,
I've reformatted properties.py with black and forced push, but I then I couldn't update the pull request, so I've closed it and opened this one. hope it's OK now.

**Comment by astrojuanlu** [comment]
No problem @imiara! Now quality is good and nothing broke, but we should add some tests so the coverage does not decrease :) Do you feel confident to use some of the existing ones as inspiration?

File: src/czml3/properties.py [hunk 1]
```diff
@@ -293,6 +293,34 @@ class BoxDimensions(BaseCZMLObject, Interpolatable):
     reference = attr.ib(default=None)
 
 
+# noinspection PyPep8Naming
+@attr.s(repr=False, frozen=True, kw_only=True)
+class Cartographicrle(BaseCZMLObject, Interpolatable, Deletable):
```
Comment (astrojuanlu) [inline src/czml3/properties.py]: Is this name OK?

Comment (idanmiara) [reply src/czml3/properties.py]: oops.. now fixed.

**Review by idanmiara — COMMENTED** [review]
Hi,
I've pushed a new test.
I'm not sure the meaning of the CI output...
could you clarify please?

**Comment by astrojuanlu** [comment]
Thanks @idanmiara , this is almost there! `isort` is complaining about the order of the imports (yes, we're _that_ picky). Just apply `$ tox -e reformat` (you need `pip install tox` first) and commit the results.

File: tests/test_rectangle_image.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+
+def save_czml_to_file(packet, out_filename):
+    czml_doc = Document([Preamble(), packet])
+    with open(out_filename, "w") as f:
+        print(czml_doc, file=f)
+
+
+def get_named_temporary_filenme(suffix="", dir_name=""):
+    filename = next(tempfile._get_candidate_names()) + suffix
```
Comment (astrojuanlu) [inline tests/test_rectangle_image.py]: MyPy is complaining about this function, which apparently does not exist?

File: tests/test_rectangle_image.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    if dir_name is None:
+        return filename
+    else:
+        if dir_name == "":
+            dir_name = tempfile._get_default_tempdir()
```
Comment (astrojuanlu) [inline tests/test_rectangle_image.py]: This doesn't exist either

**Review by astrojuanlu — CHANGES_REQUESTED** [review]
It seems to me that you are reimplementing part of the `tempfile` module itself, perhaps using some private functions, and that's why MyPy is complaining. I really like the tests here, if you can find a simpler way of making the last two, we would be in a position to merge this!

**Comment by astrojuanlu** [comment]
For example https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile

**Comment by idanmiara** [comment]
Hi,
I don't know MyPy, but these function do exist in my environment. Maybe it doesn't like that they are underscored prefixed. Anyway I did find that there is a built in function that I can use instead, so no problem. tempfile.NamedTemporaryFile creates a temp file, I just needed a temp filename - tempfile.mktemp() does that.
Could you tell me where do you find these reports or how do I run all these tests myself before pushing it? I've tried to look into the codecov report but I couldn't find what's going on there.

**Comment by astrojuanlu** [comment]
Hi @idanmiara - to see these checks, you can click here:

Screenshot_2020-02-20 Added support for CZML rectangles by idanmiara · Pull Request #58 · poliastro czml3

And see the details of every step.

To check if the quality step will pass you don't need mypy either, doing `tox -e quality` will run `isort`, `mypy`, `black` and `flake8`. Also, if you do `tox -e test`, all the tests will be run.

**Review by astrojuanlu — APPROVED** [review]
I don't want to annoy you with more reviews, for I think this is good enough. Thank you very much! :sparkles:

**Comment by idanmiara** [comment]
@Juanlu001 Thank you very much for spending the time with this code review and for merging it!
When do you plan to make new releases and publish to PyPI and to Conda ? Do you do it regularly ?

**Comment by astrojuanlu** [comment]
I made a couple of fixes: #59 and #60 (so it ended up being called `Rectangle` like the original at https://github.com/AnalyticalGraphicsInc/czml-writer/wiki/Rectangle).

For the moment I don't do releases regularly, so now that I'm at it I'll make a new one with these changes :) Stay tuned!

**Comment by astrojuanlu** [comment]
czml3 0.5.3 available with `Rectangle` :tada:

**Pull request merged**